app/src/main/java/com/example/androidapp/DatabaseController.java

```diff
@@ -239,6 +239,36 @@ public static int loadStepsBetweenDates(Context context, String date1, String da
         Log.d("STORED TIMESTAMPS: ", String.valueOf(dates.size()));
         return dates.size();
     }
+    public static Map<String, Integer> loadStepsBy30Day(Context context,String date1, String date2){
```

Comment:
This method doesn't perform what the name suggests: it returns the steps done by the user between two day, not in the last 30 days. Change the name.

Moreover, once you have this method, what's the meaning of the one called loadStepsBetweenDates()? Which, despite the name, it actually count the steps...? I would say to remove loadStepsBetweenDates() because you can count the steps "client-side" from the map returned by this method.

Reply:
thats very smart, i didn't noticed, solved